### ISSUE TITLE
Non-indexed files

### ISSUE THREAD

**Issue by 54D-R0807** [vanilla-music/vanilla#593]
Hi,
some files will not be indexed by Vanilla Music Player.
I had this problem for quiet a while. I tried changing from internal to external SD card and vice versa.
Then I changed the filesystem type from FAT to EXT4 and to exFAT (external SD card), hoping this would solve the issue.
I triggered the build in system media scanner (via developer settings) many times and also the Vanilla Music Player build in media scanner, with all possible options.

But some tracks just seem to be ignored, by the media indexer. Sometimes, files which have already been indexed, disappear. The only way to play them, is to long press on them and then either play them, or add them to the queue. The odd thing is, they do appear in the current queue, but cannot be added to an existing playlist. Also they appear with an empty name. Only the comma sign (" , ") is displayed in the current queue.

If you need any further information for trouble shooting, please let me know.

Thanks in advance.

![screenshot_2017-05-12-14-51-37](https://cloud.githubusercontent.com/assets/27963993/25999120/3a99e7b4-3724-11e7-8c77-fe841dc0b559.png)
![screenshot_2017-05-12-14-51-50](https://cloud.githubusercontent.com/assets/27963993/25999124/3a9e8170-3724-11e7-95ee-c14316997a6c.png)
![screenshot_2017-05-12-14-51-57](https://cloud.githubusercontent.com/assets/27963993/25999123/3a9e6258-3724-11e7-9bb9-8483a24ffb68.png)
![screenshot_2017-05-12-14-52-18](https://cloud.githubusercontent.com/assets/27963993/25999121/3a9b83d0-3724-11e7-8395-fa76dae91585.png)
![screenshot_2017-05-12-14-52-41](https://cloud.githubusercontent.com/assets/27963993/25999122/3a9e108c-3724-11e7-9049-29e9e3790f77.png)
![screenshot_2017-05-12-15-15-05](https://cloud.githubusercontent.com/assets/27963993/25999613/140fa758-3726-11e7-8116-a4197f46c83f.png)

**Comment by kirbyfan64** [comment]
Have you tried using the built-in tag reader instead?

**Comment by 54D-R0807** [comment]
I did, did not change anything.

**Comment by 54D-R0807** [comment]
Over the weekend I will try and find other songs, which are not being indexed and look for similarities (like same used encoding library etc.).
Maybe there is something they all have in common.
Copying the file in question and renaming it did not solve the issue. So there is probably something with the file, the player/system does not like.

**Comment by Adonai** [comment]
Adrian will want to take this file for inspection, can you please upload it somewhere?

**Comment by adrian-bl** [comment]
Most likely we are not able to read the header or android reports the file to not (only) contain audio.

You can e-mail me the file to adrian@blinkenlights.ch or upload it

**Comment by PanderMusubi** [comment]
What is exactly the built-in tag reader? This is cunfusing for some people and I keep on forgetting which is which. Please, change the text to "built-in tag reader" or "built-in Vanilla Music tag reader"

**Comment by 54D-R0807** [comment]
Please don't judge my taste of Music. ;-)
[test.mp3.zip](https://github.com/vanilla-music/vanilla/files/1000259/test.mp3.zip)

**Comment by PanderMusubi** [comment]
Yesterday, I also found myself in troubles again with adding new music to Vanilla Music. I use latest version EasyTag in latest version Ubuntu with ID3 version 2.4. All went fine before but now not unfortunately. I think Vanilla Music is one of the best music apps and I am determined to fix this as more using might experience this. Especially new users will not file a bug report of this happens on the first try and will uninstall the app. Here is a file I am having troubles with: 
[pander.zip](https://github.com/vanilla-music/vanilla/files/1000643/pander.zip)

**Comment by PanderMusubi** [comment]
Hmmm, I am afraid ID3 v 2.4.0 is not supported properly and 2.3.0 is supported.

**Comment by 54D-R0807** [comment]
@PanderMusubi The file seems to be damaged. There is nothing but noise.

**Comment by PanderMusubi** [comment]
Ah, the file is white noise. :-) I made it myself in order not to violate any copyrights. It only concerns the ID3 tags.

**Comment by 54D-R0807** [comment]
Your file works on my system. I had to rescan the sdcard though, but that is because i copied the file via kdeconnect. That never works without a rescan.

**Comment by PanderMusubi** [comment]
I use Syncthing, I can find the files via the file browser and play them but the ID3 doesn't processed properly, even though how many and which rescans and reboots I do. The deice I have is a One Plus One with Cyanogen 13.1.2 Android 6.0.1. A rescan does count the newly added files but places them under `<No Album>` and `<No Artist>`.

**Comment by adrian-bl** [comment]
@54D-R0807 I had a look at the uploaded file:

The header looks like this:
`
$ hexdump -C test.mp3|head -1
00000000  ff f3 c0 64 00 00 00 00  00 00 00 00 00 00 00 00  |...d............|
`

So in binary, that is:
`
perl -e 'print unpack("B*", pack("H*", "FFF3"))."\n"'
1111111111110011
`

Which means that this is an MPEG 2.5 layer 1 file (See: http://www.datavoyage.com/mpgscript/mpeghdr.htm) - and we don't support this type of file.

What kind of software did you use to produce the file? Lame defaults to ISO-13818-3 files.
But anyway: The simplest way to fix this is to tag them with an ID3v2 tag: This will make it possible for vanilla music to correctly guess the file type.

**Comment by 54D-R0807** [comment]
@adrian-bl You seem to really know what your doing! ;-)

I created that file with Audacity. I uses (as far as I can remember) the default export settings for MP3 files.
After reading your post, I used "Sound Converter" to convert it and pushed it to my phone. Now the file is being read by Vanilla Music.

Before I would like to mark this topic as closed, I have noticed over the past week, that many (not all) tracks in my library are duplicated in playlist (see screenshots). It does not matter, if I use the integrated Tag-reader or not. Should I file a new issue, or is this some how related? How can I assist with further investigation?

![screenshot_2016-11-16-15-23-41](https://cloud.githubusercontent.com/assets/27963993/26542278/9a1b1450-4459-11e7-9206-da821daed837.png)
![screenshot_2017-05-29-02-38-25](https://cloud.githubusercontent.com/assets/27963993/26542279/9a2656bc-4459-11e7-8f14-cac6b91e146f.png)

**Comment by PanderMusubi** [comment]
I have noticed switching back to ID3v2.3 fixes the <No Artist> et cetera and non-indexed files problem for me. Obviously, there is something amiss which is very hard to pinpoint as the file I shared here is not a problem for others with also Android 6 and the latest version of Vanilla Music. Also this answer https://sourceforge.net/p/kid3/bugs/159/ and the fact that I have other computers I play my music on has motivated me to go back to ID3v2.3 for backwards comparability.

Even the default GNU/Linux library for ID3, id3lib, is not supporting ID3v2.4 yet. Please vote here https://sourceforge.net/p/sox/bugs/291/ to help this move forward.

**Comment by adrian-bl** [comment]
> tracks in my library are duplicated in playlist (see screenshots)

In the playback view: long-press on the artist name: This should bring up additional information, including the file path: Is the path the same for both duplicates?

**Comment by PanderMusubi** [comment]
Pfff, the proper rescan appeared only after two weeks. Is it possible that the app can trigger an even harder reset/rescan? Currently, it does not suffice.

**Comment by adrian-bl** [comment]
The duplicate track issue was handled in #609 - closing this bug as the issue is fixed in the latest release.
